src/model.js

```diff
@@ -1283,16 +1283,21 @@ ${associationOwner._getAssociationDebugList()}`);
           const references = currentAttribute.references;
           if (currentAttribute.references) {
             const database = this.sequelize.config.database;
-            const schema = this.sequelize.config.schema;
+            const schema = tableName.schema;
+            const foreignReferenceSchema = currentAttribute.references.model.schema;
+            const foreignReferenceTableName = typeof references.model === 'object'
```

Comment:
`references.model` can also be a model, on which `getTableName` should be called to get the table name (could you add a test for this as well, can be done by defining an attribute with the `references` option set manually).

There is a function `isModelStatic` which can be used to determine whether `references.model` is a model class

Reply:
We have ```attribute.references.model = attribute.references.model.getTableName()``` that replace model to object.